### DISCUSSION TITLE
Confusing properties for BT-76-Lot

### DISCUSSION THREAD

**Discussion by JulianMa** [OP-TED/eForms-SDK#761] [Q&A] [answered]
Hello,

right now, `BT-76-Lot` ist optional for all notices and not conditionally forbidden, which can cause errors in dynamic schematron in specific configurations.

Given the following Values
```
BT-76-Lot = Test-Input
BT-761-Lot = true
```
the XML representation will yield the following Node:
```
<cac:TendererQualificationRequest>
	<!--BT-761-->
	<cbc:CompanyLegalFormCode listName="required">true</cbc:CompanyLegalFormCode>
	<!--BT-76-->
	<cbc:CompanyLegalForm languageID="ENG">Test-Input</cbc:CompanyLegalForm>
</cac:TendererQualificationRequest>
```
This is expected and will not result in any validation issues. By now removing the value from `BT-761-Lot` (which is also optional), the XML will look like this:
```
<cac:TendererQualificationRequest>
	<!--BT-76-->
	<cbc:CompanyLegalForm languageID="ENG">Test-Input</cbc:CompanyLegalForm>
</cac:TendererQualificationRequest>
```
This itself is valid by schema, but will cause the dynamic schematron validation of the neighbouring node to fail with messageId
`rule|text|BR-BT-00071-0072` (specific message for subType 16, but not limited to it ).

The conflicting Rule:
```
<rule context="/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingTerms/cac:TendererQualificationRequest[not(cbc:CompanyLegalFormCode)][not(cac:SpecificTendererRequirement/cbc:TendererRequirementTypeCode[@listName='missing-info-submission'])][$noticeSubType = '16']">
		<assert id="BR-BT-00071-0072" role="ERROR" diagnostics="ND-LotReservedParticipation_BT-71-Lot" test="count(cac:SpecificTendererRequirement[cbc:TendererRequirementTypeCode/@listName='reserved-procurement']/cbc:TendererRequirementTypeCode[@listName='reserved-procurement']) &gt; 0">rule|text|BR-BT-00071-0072</assert>
	</rule>
```
will also wrongly select the given XML Node, which causes the Rule to fail. 
Extending the Rule-Context with a [not(cbc:CompanyLegalForm)] would fix this, but the better way of handling would be, to introduce a Forbidden Rule for BT-76-Lot, when BT-761-Lot is not set, or to reflect it even better, not set to `true`.

**Comment by YvesJo** [comment, marked as answer]
Hi Julian,
Thanks for sharing you experience, analysis and suggestions.
As you underline, BT-76 and BT-761 are currently just optional while there exists an interdependency (Description BT-76 may only be and must be provided when the indicator BT-761 is set to "true".) 
There is a set of conditional rules already defined (BR-BT-00076-0012 ...) and deactivated since SDK 1.3. Application of these rules is part of our ongoing process of progressive rules reactivation. The rules in question shall be part of SDK 1.11.
KR,
